elements/drawtools/src/helpers.js

```diff
@@ -0,0 +1,41 @@
+import * as ol from "ol/style.js";
```

Comment:
Until now, we haven't actually imported any OL modules in the drawtools, and we only had it as dev dependency for type checking.
Since OL supports [flat styles](https://openlayers.org/en/latest/apidoc/module-ol_style_flat.html), I think this could be easily replaced with a JSON style definition that doesn't need an actual module import.

Reply:
@silvester-pari 
We cannot use `FlatStyle` in `setStyle` of individual `features`.  We need to use `StyleLike`
So. should I add `ol` to dependency? 

<img width="48%" alt="Screenshot 2023-11-01 at 5 13 12 PM" src="https://github.com/EOX-A/EOxElements/assets/10809211/0149b4fe-9592-44bb-b303-c051c5868775"> <img width="48%" alt="Screenshot 2023-11-01 at 5 13 41 PM" src="https://github.com/EOX-A/EOxElements/assets/10809211/4ce50e3e-7979-48bb-8dd6-ba7e4f0f2f68">

Reply:
You're correct, I wasn't aware that `FlatStyles` can only be applied to layers. One other option would be to set a property to the `Feature` and have a conditional styling on the layer level, for example:

```
// list.js
@mouseover=${() => {
  if (this.selectedFeatureIndex === i) return;
  feature.set("_highlighted", true);
}}
@mouseout=${() => {
  if (this.selectedFeatureIndex === i) return;
  feature.unset("_highlighted");
}}

// layer definition
{
  "type": "Vector","id": "draw","source": {"type": "Vector"},
  "style": [
    {
      "filter": ["get", "_highlighted"],
      "style": {
        "fill-color": "red"
      }
    },
    {
      "else": true,
      "style": {
        "fill-color": "blue"
      }
    }
  ]
},
```

Like this, the "highlighted" state is actually stored in the feature (which could come in handy for other use cases as well), but not the styling, so we would have a separation between features and styling.

If done like this, I'd suggest applying a default layer style to the vector layer if none is provided by the developer, but let the developer also define a custom feature highlighting style.

When using the drawtools in conjunction with `eox-map` we could also use the "programmatic selecting" functionality as introduced here: https://github.com/EOX-A/EOxElements/pull/266. This would require setting up an interaction on the map (e.g. a hover or a click interaction, useful for selecting features on the map directly), and then call `selectInteraction.highlightById([id])`.

Both solutions would be fine and wouldn't need the `ol` dependency. From my current understanding, the first solution would be more generic and would allow using the drawtools also with other OL maps, whereas the second solution would couple the drawtools closer to the `eox-map`, but would pave the way for further feature selection evolution.